File: 백준/Bronze/9046. 복호화/복호화.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        sc.nextLine();
        
        for (int t = 0; t < N; t++) {
            String S = sc.nextLine().replace(" ", "");
            int[] alpha = new int[26];
            
            for (int i = 0; i < S.length(); i++) {
                char c = S.charAt(i);
                    alpha[c - 'a']++;
            }
            
            int maxN = 0;
            int dap = -1;
            for (int i = 0; i < 26; i++) {
                if (alpha[i] > maxN) {
                    maxN = alpha[i];
                    dap = i;
                }
            }
            
            int cnt = 0;
            for (int i = 0; i < 26; i++) {
                if (alpha[i] == maxN) {
                    cnt++;
                }
            }
            
            if (cnt > 1) {
                System.out.println("?");
            } else {
                System.out.println((char) (dap + 'a'));
            }
        }
        sc.close();
    }
}

File: 백준/Bronze/9046. 복호화/복호화.c
#include <stdio.h>
#include <string.h>

int main() {
    int N;
    scanf("%d", &N);
    getchar();

    while (N--) {
        char S[1001];
        fgets(S, 1001, stdin);
        
        int alpha[26] = {0};
        
        for (int i = 0; S[i] != '\0'; i++) {
            if (S[i] != ' ') {
                alpha[S[i] - 'a']++;
            }
        }
        
        int maxN = 0;
        for (int i = 0; i < 26; i++) {
            if (alpha[i] > maxN) {
                maxN = alpha[i];
            }
        }

        int cnt = 0;
        for (int i = 0; i < 26; i++) {
            if (alpha[i] == maxN) {
                cnt++;
            }
        }

        if (cnt > 1) {
            printf("?\n");
        } else {
            for (int i = 0; i < 26; i++) {
                if (alpha[i] == maxN) {
                    printf("%c\n", i + 'a');
                    break;
                }
            }
        }
    }
    return 0;
}